keyboards/handwired/onekey/keymaps/digitizer/keymap.c

```diff
@@ -0,0 +1,38 @@
+ /* Copyright 2021 QMK 
+  * 
+  * This program is free software: you can redistribute it and/or modify 
+  * it under the terms of the GNU General Public License as published by 
+  * the Free Software Foundation, either version 2 of the License, or 
+  * (at your option) any later version. 
+  * 
+  * This program is distributed in the hope that it will be useful, 
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+#include QMK_KEYBOARD_H
+
+#include "digitizer.h"
+
+#include "math.h"
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {LAYOUT_ortho_1x1(KC_A)};
+
+uint32_t timer = 0;
+
+void matrix_scan_user() {
+    if (timer_elapsed32(timer) < 200) {
+        return;
+    }
+    
+    timer               = timer_read32();
+    digitizer_t digitizer;
+    digitizer.x         = 0.5 - 0.2 * cos(timer_read() / 250. / 6.28);
+    digitizer.y         = 0.5 - 0.2 * sin(timer_read() / 250. / 6.28);
+    digitizer.tipswitch = 0;
+    digitizer.inrange   = 1;
+    digitizer_set_report(digitizer);
```

Comment:
Shouldn't this have the send? 
```suggestion
    digitizer_set_report(digitizer);
    digitizer_send();
```

Reply:
pointing_device_task() does call pointing_device_send(); so it seems calling send explicitly is not striclty required. It makes sense to let the firmware chose when to actually send the report I guess, so I was aiming towards that API. Now this approach is intertwined with the previous review regarding using digitizer_task to update the report : either digitizer_task implements some internal behavior, then replacing it shouldn't be a solution for the end user. Or digitizer_task is public api, then it shouldn't implement internal behavior that the user could fail to reproduce and wonder why it's not working. One solution to this would be a firmware-level digitizer task and a keyboard/user-level digitizer_task, which ends up reimplementing the multi-level matrix_scan routine. 
Adding to that, the most common use case for digitizer will be sending them on keystrokes, so opening up the digitizer_task as a public API doesn't seem that useful.
Basically the two options are : 

- digitizer_task does nothing, the user has to send explicitly (which is how pointing device is advised to be used, but not how it's implemented)
- digitizer_task sends updates when one is detected, only the set_report function is useful.